src/NumField/Subfields.jl

```diff
@@ -166,40 +203,80 @@ function _subfield_primitive_element_from_basis(K::AbsSimpleNumField, as::Vector
     return a
   end
 
-  @vprintln :Subfields 1 "Trying combinations of elements in the basis"
-  # Notation: cs the coefficients in a linear combination of the as, ca the dot
-  # product of these vectors.
-  cs = ZZRingElem[rand(ZZ, -2:2) for n in 1:dsubfield]
-  k = 0
-  s = 1
-  first = true
-  a = one(K)
-  I = t2(a)
-  while true
-    s += 1
-    ca = sum(c*a for (c,a) in zip(cs,as))
-    b = _block(ca, rt, ap)
-    if length(b) == dsubfield
-      t2n = t2(ca)
-      if first
-        a = ca
-        I = t2n
-        first = false
-      elseif t2n < I
-        a = ca
-        I = t2n
-      end
-      k += 1
-      if k == 5
-      	@vprintln :Subfields 1 "Primitive element found"
-        return a
+  @vprintln :Subfields 1 "Finding combination of elements"
+
+  pe = as[1]
+  cur_b = all_b[1]
+  for i=2:length(as)
+    if issubset(all_b[i][1], cur_b[1])
+      continue
+    end
+    cur_b = Vector{Int}[intersect(x, y) for x in cur_b for y in all_b[i]]
+    cur_b = Vector{Int}[x for x in cur_b if length(x) > 0]
+    j = 1
+    while block_system(pe + j*as[i], C) != c
```

Comment:
c?

Reply:
cur_b - it was renamed a couple of times